### DISCUSSION TITLE
System Header variable overlap

### DISCUSSION THREAD

**Discussion by youngmi97** [ANLAB-KAIST/KENSv3#61] [Project Q&A 2. Transport Layer (TCP)]
Hi, 
Im facing an error during build which seems to suggest that the variable declaration is happening elsewhere.
From what I understand, Internal linkage refers to everything only in scope of a translation unit, and the static variable calls for such internal linkage. But in this case the linkage seems to be mixed up.

No files other than TCPAssignment.cpp/hpp were touched.

E_Host.hpp
```cpp

class SystemCallInterface {
 public:
  static constexpr int AF_INET = 2;
  static constexpr int IPPROTO_TCP = 6;
  static constexpr int IPPROTO_UDP = 17;


```

<img width="1187" alt="Screen Shot 2022-04-07 at 5 01 45 AM" src="https://user-images.githubusercontent.com/38910144/162060966-f0f86a84-0ee8-4f25-94e4-b7fa2a78896d.png">

The following are my build configurations using VScode:
<img width="951" alt="Screen Shot 2022-04-07 at 5 02 10 AM" src="https://user-images.githubusercontent.com/38910144/162061032-c8ee4b8b-fa7d-4d1f-928d-dd1b793963b2.png">

**Comment by Pusnow** [comment]
You may have to *uninclude* system's socket headers in `TCPAssignment.cpp/hpp `.
